src/borrow_tracker/tree_borrows/perms.rs

```diff
@@ -271,7 +343,7 @@ pub mod diagnostics {
         pub fn summary(&self) -> &'static str {
             assert!(Self::is_possible(self.0, self.1));
             match (self.0, self.1) {
-                (_, Active) => "an activation",
+                (_, Active) => "the first write to a 2-phase borrowed pointer",
```

Comment:
"pointer" often indicates "raw pointer" so we should probably avoid it here.
```suggestion
                (_, Active) => "the first write to a 2-phase borrowed mutable reference",
```